server/plugins/renderer/renderer.module.js

```diff
@@ -429,7 +419,23 @@ internals.getHeaders = (request, stencilOptions = {}, stencilConfig) => {
         headers['stencil-store-url'] = request.app.storeUrl;
     }
 
-    return { ...request.headers, ...headers };
+    return { ...request.headers, ...headers, ...extraHeaders };
+};
+
+/**
+ * Process headers from Fetch response
+ *
+ * @param {Headers} headers
+ * @returns {object}
+ */
+internals.processResHeaders = (headers) => {
+    if (headers.get('set-cookie')) {
+        // When there are several values for the same header headers.get() returns
+        //  an array joined with comma which is wrong for cookies, so we use the initial raw Array
+        // https://github.com/node-fetch/node-fetch/issues/251#issuecomment-428143940
+        // eslint-disable-next-line no-param-reassign
+        headers.raw()['set-cookie'] = utils.stripDomainFromCookies(headers.raw()['set-cookie']);
```

Comment:
Maybe it's time to consider another library? :)

Reply:
Maybe😄 (but in a separate PR)
Which one do you recommend? Axios?

Reply:
yeah, I enjoy working with axios

Reply:
@junedkazi if you agree, I'll create a ticket for replacing fetch with Axios since Axios figured out to be more handy

Reply:
Let's create a ticket and we can switch over to axios.

Reply:
@jairo-bc , @junedkazi , ticket: https://jira.bigcommerce.com/browse/STRF-8745